### PR TITLE
Work on database classes completed

### PR DESCRIPTION
**Pull request by rrivera992011**
Need review in case I missed something in the writing

### CODE REVIEW

**Comment by ftranlaba** [comment]
Hello Rafael, nice work. There is an issue though. There is a lot of duplicated setters and getters. Can you apply the abstract classes found in the hierarchy diagram? So, the abstract classes will contain the "shared field" between concrete classes. Through inheritance, the "shared field" can be declared private and in the child class's constructor, you can use the keyword super to initialize the shared field. Next, the setter/getter does not need to be rewritten in the child classes because the abstract classes will have public getters and setters which is passed off to the child classes.

**Comment by Moussa-Mikhail** [comment]
I second Francis' suggestion. A uniform way to access the id field across all models would reduce the amount of work I have to do to make the DAOs work.

File: src/main/java/datamodels/AccountIdInfo.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package datamodels;
+
+public class AccountIdInfo extends IdInfo{
```
Comment (ftranlaba) [inline src/main/java/datamodels/AccountIdInfo.java]: abstract class does not need constructor

File: src/main/java/datamodels/AccountAccessHistory.java [hunk 2]
```diff
@@ -2,26 +2,24 @@
 
 import java.sql.Timestamp;
 
-public class AccountAccessHistory {
-    private int accountAccessHistoryId;
-    private int accountId;
+public class AccountAccessHistory extends IdInfo {
```
Comment (ftranlaba) [inline src/main/java/datamodels/AccountAccessHistory.java]: This class does not follow the inheritance hierarchy and name

File: src/main/java/datamodels/AccountIdInfo.java (hunk 1 above)
Comment (Moussa-Mikhail) [reply src/main/java/datamodels/AccountIdInfo.java]: I think we're unclear on what the purpose of this class is supposed to be. Also if it is meant to be abstract then declare it with the abstract keyword.

Comment (ftranlaba) [reply src/main/java/datamodels/AccountIdInfo.java]: my apologies, an abstract class should have a constructor in order to initialize fields and call super().

File: src/main/java/datamodels/AccountIdInfo.java [hunk 3]
```diff
@@ -1,4 +1,18 @@
 package datamodels;
 
-public class AccountIdInfo extends IdInfo{
+public class AccountIdInfo extends IdInfo {
```
Comment (ftranlaba) [inline src/main/java/datamodels/AccountIdInfo.java]: this should be abstract

**Pull request merged**